### PR TITLE
Adds parsing ability to new StorageKey classes

### PR DESCRIPTION
**Pull request by csilvestrini**

### CODE REVIEW

File: src/runtime/storageNG/drivers/firebase.ts [hunk 1]
```diff
@@ -38,6 +38,18 @@ export class FirebaseStorageKey extends StorageKey {
...
+
+  static fromString(key: string): FirebaseStorageKey {
+    const match = key.match(/^firebase:\/\/([^.]+)\.([^:]+):([^/]+)\/(.*)$/);
+    if (!match) {
+      throw new Error(`Not a valid FirebaseStorageKey: ${key}.`);
+    }
+    const projectId = match[1];
+    const domain = match[2];
+    const apiKey = match[3];
+    const location = match[4];
```
Comment (mykmartin) [inline src/runtime/storageNG/drivers/firebase.ts]: Suggestion: `const [_, projectId, domain, apiKey, location] = match;`

File: src/runtime/storageNG/drivers/volatile.ts [hunk 2]
```diff
@@ -32,6 +32,16 @@ export class VolatileStorageKey extends StorageKey {
...
     return new VolatileStorageKey(this.arcId, `${this.unique}/${component}`);
   }
+
+  static fromString(key: string): VolatileStorageKey {
+    const match = key.match(/^volatile:\/\/([^/]+)\/(.*)$/);
+    if (!match) {
+      throw new Error(`Not a valid VolatileStorageKey: ${key}.`);
+    }
+    const arcId = ArcId.fromString(match[1]);
+    const unique = match[2];
```
Comment (mykmartin) [inline src/runtime/storageNG/drivers/volatile.ts]: Similar here.

File: src/runtime/storageNG/drivers/firebase.ts (hunk 1 above)
Comment (csilvestrini) [reply src/runtime/storageNG/drivers/firebase.ts]: Good idea, done!

File: src/runtime/storageNG/drivers/volatile.ts (hunk 2 above)
Comment (csilvestrini) [reply src/runtime/storageNG/drivers/volatile.ts]: Done

**Pull request merged**